### PR TITLE
Add CodeQL Analysis workflow

### PR DESCRIPTION
**Pull request by fisker**
## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: website/static/service-worker.js [hunk 1]
```diff
@@ -15,33 +17,33 @@ toolbox.precache([
   "lib/parsers-location.js",
   ...plugins,
   "playground.js",
-  "https://unpkg.com/sw-toolbox@3.6.0/sw-toolbox.js",
+  "https://cdnjs.cloudflare.com/ajax/libs/sw-toolbox/3.6.1/sw-toolbox.js",
```
Comment (kachkaev) [inline website/static/service-worker.js:20]: You probably know that but still worth flagging that `sw-toolbox` is deprecated:
https://github.com/GoogleChromeLabs/sw-toolbox

Also 3.6.1 does not exist on NPM (https://www.npmjs.com/package/sw-toolbox/v/3.6.1), only on GitHub (https://github.com/GoogleChromeLabs/sw-toolbox/releases/tag/v3.6.1). Not sure where CloudFlare pulls the artifact from, but it seem to be doing that correctly 🤔

File: website/pages/playground/index.html [hunk 2]
```diff
@@ -28,45 +28,154 @@
...
+      referrerpolicy="no-referrer"
+    ></script>
     <link rel="stylesheet" href="/separate-css/playground.css" />
 
     <!-- Global site tag (gtag.js) - Google Analytics -->
     <script
       async
       src="https://www.googletagmanager.com/gtag/js?id=UA-111350464-1"
+      crossorigin="anonymous"
+      referrerpolicy="no-referrer"
```
Comment (kachkaev) [inline website/pages/playground/index.html:178]: [gtag.js docs](https://developers.google.com/tag-platform/gtagjs/install) don’t have these extra attributes. They [might break tracking](https://support.google.com/tagmanager/thread/98360689/script-src-for-gtag-and-crossorigin-anonymous-does-not-load-the-js-file-from-cdn?hl=en), but I’m not 100% sure.

Comment (fisker) [reply website/pages/playground/index.html]: I don't think it's true. But let's trust google "Don't be evil".

File: website/static/service-worker.js (hunk 1 above)
Comment (fisker) [reply website/static/service-worker.js]: Downgraded to 3.6.0

Comment (fisker) [reply website/static/service-worker.js]: https://github.com/prettier/prettier/issues/8763#issuecomment-663385515

**Pull request merged**